rrmngmnt/nmcli.py

```diff
@@ -0,0 +1,644 @@
+            command += " ipv6.addresses {ipv6_addr}".format(ipv6_addr=ipv6_addr)  # noqa: E501
+        if ipv6_gw:
+            command += " ipv6.gateway {ipv6_gw}".format(ipv6_gw=ipv6_gw)
+        return command
+
+    @staticmethod
+    def _common_options_builder(
+        con_type, con_name, ifname, auto_connect=None, save=None
+    ):
+        """
+        Generates a string containing common options for the nmcli tool.
+
+        Args:
+            con_type (str): the connection type.
+            con_name (str): the created connection's name.
+            ifname (str): the interface name to use.
+            auto_connect (bool): True to connect automatically, or False for
+                manual.
+            save (bool): True to persist the connection, or False.
+
+        Returns:
+            str: a common options string.
+        """
+        common_options = (
+            "type {type} " "con-name {con_name} " "ifname {ifname} "
+        ).format(type=con_type, con_name=con_name, ifname=ifname)
+        if auto_connect is not None:
+            common_options += "autoconnect {val}".format(
+                val="yes" if auto_connect is True else "no"
+            )
+        if save is not None:
+            common_options += "save {val}".format(val="yes" if save is True else "no")  # noqa: E501
+
+        return common_options
+
+    def _nmcli_con_cmd_builder(
+        self,
+        operation,
+        con_name,
+        con_type=None,
+        ifname=None,
+        auto_connect=None,
+        save=None,
+        ipv4_method=None,
+        ipv4_addr=None,
+        ipv4_gw=None,
+        ipv6_method=None,
+        ipv6_addr=None,
+        ipv6_gw=None,
+        type_options=None,
+    ):
+        """
+        Builds an nmcli command.
+
+        Args:
+            operation (str): the operation to perform. e.g: "add", "delete" ...
+            con_name (str): the created connection's name.
+            con_type (str): the connection type. e.g: "ethernet", "bond" ...
+            ifname (str): the interface name to use.
+            auto_connect (bool): True to connect automatically, or False for
+                manual.
+            save (bool): True to persist the connection, or False.
+            ipv4_method (str): setting method.
+                Available methods: auto, disabled, link-local, manual, shared.
+            ipv4_addr (str): a static address.
+            ipv4_gw (str): a gateway address.
+            ipv6_method (str): setting method.
+                Available methods: auto, dhcp, disabled, ignore, link-local,
+                manual, shared.
+            ipv6_addr (str): a static address.
+            ipv6_gw (str): a gateway address.
+            type_options (dict): type specific options. e.g: {"mtu": "1500"}.
+
+        Returns:
+            str: an nmcli command.
+        """
+        command = "nmcli con {operation}".format(operation=operation, con=con_name)  # noqa: E501
+
+        if operation == "delete":
```

Comment:
if operation == "delete" or operation == "modify":
    command += " {con}".format(con=con_name)

elif operation == "add":
    command += " {common}".format(
                common=self._common_options_builder(
                    con_type=con_type,
                    con_name=con_name,
                    ifname=ifname,
                    auto_connect=auto_connect,
                    save=save,
                )
            )
    command += self._ip_options_builder(
                ipv4_addr=ipv4_addr,
                ipv4_gw=ipv4_gw,
                ipv4_method=ipv4_method,
                ipv6_addr=ipv6_addr,
                ipv6_gw=ipv6_gw,
                ipv6_method=ipv6_method,
            )

 if type_options:
                for k, v in type_options.items():
                    command += " {k} {v}".format(k=k, v=v)